scripts/calculate_aligned_extent.sql

```diff
@@ -0,0 +1,59 @@
+create or replace function calculate_aligned_extent(facility_polygon_id integer, region_id integer)
+returns text as $$
+declare
+  isochrone_xmin double precision;
+  isochrone_ymin double precision;
+  isochrone_xmax double precision;
+  isochrone_ymax double precision;
+  region_xmin double precision;
+  region_ymin double precision;
+  region_xmax double precision;
+  region_ymax double precision;
+  unaligned_target_minx double precision;
+  unaligned_target_miny double precision;
+  unaligned_target_maxx double precision;
+  unaligned_target_maxy double precision;
+  srs_block_length double precision;
+  minx_blocks double precision;
+  target_minx double precision;
+  maxx_blocks double precision;
+  target_maxx double precision;
+  miny_blocks double precision;
+  target_miny double precision;
+  maxy_blocks double precision;
+  target_maxy double precision;
+  aux double precision;
+begin
+  select into isochrone_xmin, isochrone_ymin, isochrone_xmax, isochrone_ymax
+  st_xmin(the_geom), st_ymin(the_geom), st_xmax(the_geom), st_ymax(the_geom)
+  from facilities_polygons
+  where id = facility_polygon_id;
+
+  select into region_xmin, region_ymin, region_xmax, region_ymax
+  st_xmin(the_geom), st_ymin(the_geom), st_xmax(the_geom), st_ymax(the_geom)
+  from regions
+  where id = region_id;
+
+  -- calculate extent based on the block size of (128x128) and the precision of 0.0008333
+  unaligned_target_minx := greatest(isochrone_xmin, region_xmin);
+  unaligned_target_miny := greatest(isochrone_ymin, region_ymin);
+  unaligned_target_maxx := least(isochrone_xmax, region_xmax);
+  unaligned_target_maxy := least(isochrone_ymax, region_ymax);
+
+  srs_block_length := 0.000833*128;
```

Comment:
Avoid hardcoding these values here. We should extract them from the raster and pass them as extra params to the function. `gdalinfo` will return this information, and can be extracted using a regex (FWIW, python, grep, sed, perl and awk are all available on the app docker image); or we can write a small cpp that uses the GDAL API to get exactly those values.